repo/packages/algebra-plugins/package.py

```diff
@@ -0,0 +1,43 @@
+from spack.package import *
+
+
+class AlgebraPlugins(CMakePackage):
+    homepage = "https://github.com/acts-project/algebra-plugins"
+    git = "https://github.com/acts-project/algebra-plugins.git"
+    list_url = "https://github.com/acts-project/algebra-plugins/tags"
+
+    version("0.17.0", commit="f4e9c1cc37ce6080234f265d121f4bc0a74bef87")
+
+    variant(
+        "cxxstd",
+        default="17",
+        values=("17", "20", "23"),
+        multi=False,
+        description="C++ standard used",
+    )
+    variant("eigen", default=False, description="Enables the Eigen plugin")
+    variant("smatrix", default=False, description="Enables the SMatrix plugin")
+    variant("vecmem", default=False, description="Enables the vecmem plugin")
+    variant("vc", default=False, description="Enables the Vc plugin")
+
+    depends_on("cmake@3.14:", type="build")
+    depends_on("vecmem@0.22.0: +cuda", when="+vecmem")
+    depends_on("eigen@3.4.0:", when="+eigen")
+    depends_on("vc@1.4.0:", when="+vc")
+    depends_on("root@6.18.0:", when="+smatrix")
+
+    patch("fix_cmake_install.patch", when="@:0.17.99", level=1)
+
+    def cmake_args(self):
+        spec = self.spec
+
+        args = [
+            self.define_from_variant("CMAKE_CXX_STANDARD", "cxxstd"),
```

Comment:
Does the `CMakePackage` base not take care of this setting itself? :thinking: We use standard CMake variables for these purposes exactly so that package managers could work with minimum effort. (I don't know whether this setting missing from https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/vecmem/package.py is indicative of the setting not being needed, or @HadrienG2 having forgotten about it there. :stuck_out_tongue:)

Reply:
It does not. This makes sense because:

1. CMake is not just a C++ build tool (sadly), and baking C++ specific features into the `CMakePackage` mixin would violate the separation of concerns; and
2. Exposing an ABI that is dependent on the C++ standard is a pretty significant anti-pattern that should be avoided, and ideally packages would not care about the C++ standards used to build their children.